screens/auth/SignUpScreen.js

```diff
@@ -51,25 +65,61 @@ export default class SignUpScreen extends React.Component {
     };
   }
 
+  setModalVisible(visible) {
+    this.setState({ modalVisible: visible });
+  }
+
   // TODO @johnathanzhou or @anniero98
   // Notifications is jank - the `_handleNotification` function doesn't even exist. Unclear to devs what the flow should be with receiving notifications
   componentDidMount() {
     // this.registerForPushNotificationsAsync();
     // this._notificationSubscription = Notifications.addListener(this._handleNotification);
   }
 
+  openRecaptcha = async () => {
+    const number = '+1'.concat(this.state.values[signUpFields.PHONENUM]);
+    const phoneProvider = new firebase.auth.PhoneAuthProvider();
+    const verificationId = await phoneProvider.verifyPhoneNumber(
+      number,
+      this.state.ref.current
+    );
+    this.setState({ verificationId });
+    this.setModalVisible(true);
+  };
+
+  verifyCode = async code => {
+    try {
+      const credential = firebase.auth.PhoneAuthProvider.credential(
+        this.state.verificationId,
+        code
+      );
+      await firebase.auth().signInWithCredential(credential);
+      this.completeSignUp();
+    } catch (err) {
+      this.setState(prevState => ({
+        errors: {
+          ...prevState.errors,
+          [signUpFields.CODE]: 'Incorrect Code.',
```

Comment:
This may be a TODO for later, but is there a way to resend the code?

Reply:
Definitely should be, I'll look into adding that sometime soon